### ISSUE TITLE
chat - generation issue

### ISSUE THREAD

**Issue by karlie38** [haven-jeon/KoGPT2-chatbot#4]
안녕하세요!

올려주신 chat 부분에서는 원래 사용자가 대화를 치면 응답이 생성되도록 하셨는데,

저는 정해진 csv 파일을 만들고, 아래처럼 한 row씩 돌면서 문장이 생성되도록 코드를 짰습니다.

그런데, 이상하게 어느 시점이 되면, 문장을 생성하지 않고 그냥 멈쳐져 있는데, 이게 어떤 이슈일지 조언을 구하고 싶습니다.

해당 에러가 생기는 문장을 보면'라단조, sehr'만 무한으로 나오고 추가 생성은 되지 않습니다.
kogpt2는 한국어 기반인데 영어가 들어가서 그런 것인지, 아니면 다른 문제가 있는 것일까요?
이걸 해결하려면 kogpt2말고 gpt2-multi language 로 가야할까요?

확인 부탁드리겠습니다!!

`with torch.no_grad():

          for i in range(len(test_data)):
                try:
                     sent = test_data['label'].iloc[i]  # text
                     q = test_data['Q'].iloc[i]  # context
                     real_a = test_data['A'].iloc[i]  # question
                     if len(sent) > 2 and len(q) > 2:
                         sent_tokens = tok(sent)
                         q_tok = tok(q)
                         a = ''
                         a_tok = []
                         while 1:
                                input_ids = torch.LongTensor([
                                                         self.vocab[U_TKN]] + self.vocab[q_tok] +
                                                         self.vocab[EOS, SENT] + self.vocab[sent_tokens] +
                                                         self.vocab[EOS, S_TKN] +
                                                         self.vocab[a_tok]).unsqueeze(dim=0)
                                pred = self(input_ids)
                                gen = self.vocab.to_tokens(
                                    torch.argmax(
                                        pred,
                                        dim=-1).squeeze().numpy().tolist())[-1]
                                if gen == EOS:
                                    break
                                a += gen.replace('▁', ' ')
                                a_tok = tok(a)
                     print(i, '번째')`

**Comment by haven-jeon** [comment]
#3 이건 해결이 된건가요?

**Comment by karlie38** [comment]
해결 

> #3 이건 해결이 된건가요?

넵 해결되었습니다!!
그리고 제가 오늘 올림 이슈도 원인을 알았습니다!
계속 반복된 것을 생성해서 나는 이슈였는데
예를 들어,
gen = '_' (즉, 빈 space)이 나오면 뒤에도 계속 '_'만 생성하기도 하고
또 계속 황산염, 황산염을 생성하게 되는 경우도 있었습니다.

**Comment by haven-jeon** [comment]
네.. 올려주신 두 이슈 모두 close 부탁드립니다.

**Comment by Jake1152** [comment]
혹시 답변 그대로 반복되는 문제는 어떻게 해결하셨나요?
